File: Plus One/Plus_One.java
class Solution {
    public int[] plusOne(int[] digits) {
        if (digits[digits.length -1] < 9){
            digits[digits.length -1] = digits[digits.length -1] + 1;
            return digits;
        }
        boolean allNines = true;
        for ( int i = 0; i < digits.length; i++ ){
            if ( digits[i] != 9 ){
                allNines = false;
            }
        }
        if ( allNines ){
            int [] ans = new int [digits.length + 1];
            ans[0] = 1;
            for ( int i = 1; i < ans.length; i++ ){
                ans[i] = 0;
            }
            return ans;
        }
        else {
            int [] ans = new int [digits.length];
            for ( int i = 0; i < digits.length; i++ ){
                ans[i] = digits[i];
            }
            ans[digits.length -1] = 0;
            for ( int i = digits.length - 2; i >= 0; i-- ){
                if (digits[i] != 9){
                    ans[i] = digits[i] + 1;
                    return ans;
                }
                else {
                    ans[i] = 0;
                }
            }
            return ans;
        }
    }
}

File: Coding Problems & My Solutions/Plus One/Plus_One.swift
class Solution {
    func plusOne(_ digits: [Int]) -> [Int] {
        var digs = digits
        var i = digs.count - 1
        while i >= 0 {
            if digs[i] < 9 {
                digs[i] = digs[i] + 1
                return digs
            } else {
                digs[i] = 0
                i -= 1
            }
        }
        var expandedArr: [Int] = [1]
        return expandedArr + digs
    }
}